var/spack/repos/builtin/packages/quantum-espresso/package.py

```diff
@@ -79,6 +75,37 @@ class QuantumEspresso(CMakePackage):
         conflicts("^openblas threads=none")
         conflicts("^openblas threads=pthreads")
 
+    # Add cuda support
+    with when("%nvhpc"):
+        variant(
+            "cuda",
+            default=False,
+            description="Build with CUDA, OpenMP on is strongly suggested for performance",
+        )
+        with when("+cuda"):
+            # GPUs are enabled since v6.6
+            conflicts("@:6.5")
+            # cuda version >= 10.1
+            # conflicts("cuda@:10.0.130")
+            # bugs with following nvhpcsdk versions
+            conflicts(
+                "%nvhpc@21.11:22.2", msg="QE-GPU has issues with these NVHPC compiler versions"
+            )
+            # bugs with nvhpc@22.3 in versions > 7.1 still present
+            with when("@develop"):
+                conflicts("%nvhpc@22.3", msg="Develop has issues with this NVHPC compiler version")
+            with when("@7.2:"):
+                conflicts("%nvhpc@22.3", msg="v > 7.1 has issues with this NVHPC compiler version")
+            # only cmake is supported
+            conflicts("~cmake", msg="Only CMake supported for GPU-enabled version")
+            # PGI support
+            conflicts("%pgi@:19.9", msg="PGI supported for v >= 19.10")
```

Comment:
If nvhpc is required, isn't this conflict always false?

Reply:
You are right. To be consistent with CMakeLists.txt we should enable also the PGI compiler, I suggest.

Reply:
No need to waste time on pgi releases.

Reply:
removed in 487d76f99948010ddf0daf8cd1eaf35a2dd9929e